### ISSUE TITLE
Linking error: prohibited cross reference from .text.internal to `__udivmoddi4' in .text.external

### ISSUE THREAD

**Issue by nsajko** [numworks/epsilon#1168]
#### Describe the bug
The ARM target build fails:

GNU Binutils is version 2.32.
GCC is version 9.2.0.

```
...
CXX     apps/apps_container_launch_default.o
LD      epsilon.elf
/usr/lib/gcc/arm-none-eabi/9.2.0/../../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.0/thumb/v7e-m+fp/hard/libgcc.a(_aeabi_ldivmod.o): in function `__aeabi_ldivmod':
(.text+0x32): prohibited cross reference from .text.internal to `__udivmoddi4' in .text.external
/usr/lib/gcc/arm-none-eabi/9.2.0/../../../../arm-none-eabi/bin/ld: (.text+0x4c): prohibited cross reference from .text.internal to `__udivmoddi4' in .text.external
/usr/lib/gcc/arm-none-eabi/9.2.0/../../../../arm-none-eabi/bin/ld: (.text+0x6e): prohibited cross reference from .text.internal to `__udivmoddi4' in .text.external
/usr/lib/gcc/arm-none-eabi/9.2.0/../../../../arm-none-eabi/bin/ld: (.text+0x8a): prohibited cross reference from .text.internal to `__udivmoddi4' in .text.external
collect2: error: ld returned 1 exit status
make: *** [build/rules.mk:37: output/release/device/n0110/epsilon.elf] Error 1
```

Note: the simulator build succeeds and mostly works fine.

#### To Reproduce
1. Execute make
2. See error

#### Expected behavior
The build to succeed.

#### Environment
 - Epsilon version is 4b9e42d522dec8c844d66617f9968d33e43fcb5a
 - The platform(s) on which the problem happens: build error

**Comment by Ecco** [comment]
Weird. Both of those symbols should be in [internal flash](https://github.com/numworks/epsilon/blob/4b9e42d522dec8c844d66617f9968d33e43fcb5a/ion/src/device/n0110/flash.ld#L105)...

**Comment by EmilieNumworks** [comment]
Hi @nsajko,
`__udivmoddi4` is part of the GCC low-level runtime library and should be in the internal flash as specified by the linker configuration file `ion/src/device/n0220/flash.ld`:

> /* Optimization */
> */libgcc.a:(.text)
> } >INTERNAL_FLASH

It does work with 8.3.1 arm-none-eabi-g++ version but I should investigate what the issue is with the version 9...

**Comment by nsajko** [comment]
Still the same error with binutils 2.33.1.

**Comment by Ecco** [comment]
Note: on macOS we're now using GCC 9.2.1, and it seems to be working just fine.

**Comment by nsajko** [comment]
Still does not build with latest epsilon, the error messages seem the same.

What environment variables do other people use when building epsilon?

I am trying to build from an Archlinux system. Arch does not patch arm-none-eabi-binutils or arm-none-eabi-gcc, so I guess maybe there is an issue that gets patched in the version you guys are running on macOS? Could you confirm if your packages are patched? Dunno where else could the issue lie.

For what it is worth, these are the Arch build scripts for arm-none-eabi-binutils and arm-none-eabi-gcc:
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/arm-none-eabi-binutils
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/arm-none-eabi-gcc

**Comment by Ecco** [comment]
We're using the [standard ARM-provided toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm).

**Comment by nsajko** [comment]
The ARM-provided toolchain successfully built Epsilon!

Now it says I have software version 12.0.0 and commit 5ec09fd. I wonder why it does not declare as 12.2.0, though?

**Comment by Ecco** [comment]
Because 12.2 is on the 12 branch. Glad you had your issue sorted!

**Comment by RedGl0w** [comment]
This issue is fixed when using official arm toolchain from their website. But it could be cool if it would be possible to compile epsilon with official arm toolchain from package managers. I saw that you weren't able to reproduce it above. If you want to, you just have to compile it on arch with arm-none-eabi-gcc from arch repository. 
> Weird. Both of those symbols should be in [internal flash](https://github.com/numworks/epsilon/blob/4b9e42d522dec8c844d66617f9968d33e43fcb5a/ion/src/device/n0110/flash.ld#L105)...

After some testing today, I was able with toolchain from arch repository to compile epsilon when removing the `*/` at the beginning of the line you indicated. I wasn't able to find anything about linkerscript and the meaning of `*/` (except about comments, which after a test clearly wasn't).
